drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -684,6 +684,41 @@ static int stm32_clock_control_init(const struct device *dev)
 
 #endif /* CONFIG_CPU_CORTEX_M4 */
 
+#if defined(CONFIG_CLOCK_STM32_PLL3_ENABLE)
+	/* Initialize PLL 3 */
+	r = get_vco_input_range(CONFIG_CLOCK_STM32_PLL3_M_DIVISOR, &vco_input_range);
+	if (r < 0) {
+		return r;
+	}
+
+	vco_output_range = get_vco_output_range(vco_input_range);
+
+	LL_RCC_PLL3FRACN_Disable();
+
+	LL_RCC_PLL3_SetM(CONFIG_CLOCK_STM32_PLL3_M_DIVISOR);
+	LL_RCC_PLL3_SetN(CONFIG_CLOCK_STM32_PLL3_N_MULTIPLIER);
+
+	LL_RCC_PLL3_SetVCOInputRange(vco_input_range);
+	LL_RCC_PLL3_SetVCOOutputRange(vco_output_range);
+
+#if defined(CONFIG_CLOCK_STM32_PLL3_P_ENABLE)
+	LL_RCC_PLL3P_Enable();
+	LL_RCC_PLL3_SetP(CONFIG_CLOCK_STM32_PLL3_P_DIVISOR);
+#endif /* CONFIG_CLOCK_STM32_PLL3_P_ENABLE */
+#if defined(CONFIG_CLOCK_STM32_PLL3_Q_ENABLE)
+	LL_RCC_PLL3Q_Enable();
+	LL_RCC_PLL3_SetQ(CONFIG_CLOCK_STM32_PLL3_Q_DIVISOR);
+#endif /* CONFIG_CLOCK_STM32_PLL3_Q_ENABLE */
+#if defined(CONFIG_CLOCK_STM32_PLL3_R_ENABLE)
+	LL_RCC_PLL3R_Enable();
+	LL_RCC_PLL3_SetR(CONFIG_CLOCK_STM32_PLL3_R_DIVISOR);
+#endif /* CONFIG_CLOCK_STM32_PLL3_R_ENABLE */
```

Comment:
Nit: IS_ENABLED can be used here.

Reply:
@erwango using `IS_ENABLED` leads to some compile errors because `*_DIVISOR` are conditionally defined, any suggestions?

Reply:
@gmarull Ha right, I was focusing on the LL_RCC_* functions. Nvm this is fine.